book/src/concepts/entity_and_component.md

```diff
@@ -0,0 +1,105 @@
+
+## EntitiesRes
+
+Even though the structure of the entity is pretty simple, entity manipulation is very sophisticated and crucial to game preformance. This is why entities are handled exclusively by the struct `EntitiesRes`. `EntitiesRes` provides two ways for creating/deleting entities:
+
+* Immediate creation/deletion, used for game setup or clean up.
+* Atomic or delayed creation/deletion, used in the game play state. It updates entities in batch at the end of each game loop.
+
+You will see how these methods are used in later chapters.
+
+
+## Declaring a component
+
+To declare a component, you first declare the relevant underlying data:
+
+```rust,no_run,noplaypen
+/// This `Component` describes the shape of an `Entity`
+enum Shape {
+    Sphere { radius: f32 },
+    RectangularPrism { height: f32, width: f32, depth: f32 },
+}
+
+/// This `Component` describes the contents of an `Entity`
+pub struct Content {
+    content_name: String,
+}
+```
+
+and then you implement the `Component` trait for them:
+
+```rust,no_run,noplaypen
+# extern crate amethyst;
+# struct Shape;
+# struct Content;
+use amethyst::ecs::{Component, DenseVecStorage};
+
+impl Component for Shape {
+    type Storage = DenseVecStorage<Self>;
+}
+
+impl Component for Content {
+    type Storage = DenseVecStorage<Self>;
+}
+```
+Defining the Storage type will determine how you store the component, but it will not initiate the storage. Storage is initiated when you register a component in `World`. You will see it in the World chapter. 
+
+## Storages
+
+There are a few storage strategies for different usage scenarios. The most commonly used types are `DenseVecStorage`, `VecStorage` and `FlaggedStorage`.
+
+* `DenseVecStorage`: Elements are stored in a contiguous vector. No empty space is left between `Component`s, allowing a lowered memory usage for big components.
+* `VecStorage`: Elements are stored into a sparse array. The entity id is the same as the index of component. If your component is small (<= 16 bytes) or is carried by most entities, this is preferable over `DenseVecStorage`.
+* `FlaggedStorage`: Used to keep track of changes of a component. Useful for caching purposes.
+
+![DenseVecStorage](../images/concepts/component_and_entity_densevecstorage.svg)
+
+![VecStorage](../images/concepts/component_and_entity_vecstorage.svg)
+
+For more information, see the [specs storage reference](https://docs.rs/specs/latest/specs/storage/index.html).
+
+There are a bunch more storages, and deciding which one is the best isn't trivial and should be done based on careful
+benchmarking. If you don't know which one you should use, `DenseVecStorage` is a good default. It will need more memory
+than `VecStorage` for pointer-sized components, but it will perform well for most scenarios.
```

Comment:
I think the general rule of thumb torkleyy gave was if you component is used in over 30% of entities, use VecStorage.